base/array.jl

```diff
@@ -1966,7 +1971,12 @@ findfirst(p::Union{Fix2{typeof(isequal),Int},Fix2{typeof(==),Int}}, r::OneTo{Int
 findfirst(p::Union{Fix2{typeof(isequal),T},Fix2{typeof(==),T}}, r::AbstractUnitRange) where {T<:Integer} =
     first(r) <= p.x <= last(r) ? 1+Int(p.x - first(r)) : nothing
 
-function findfirst(p::Union{Fix2{typeof(isequal),T},Fix2{typeof(==),T}}, r::StepRange{T,S}) where {T,S}
+findfirst(p::Union{Fix2{typeof(isequal),T},Fix2{typeof(==),T}}, r::StepRange{T,S}) where {T,S} =
+    _findfirst(p, RangeStepStyle(r), r)
+
+_findfirst(p, ::RangeStepStyle, r) = _findfirst(p, r)
+
+function _findfirst(p::Union{Fix2{typeof(isequal)},Fix2{typeof(==)}}, ::RangeStepRegular, r::StepRange{T,S}) where {T,S}
```

Comment:
Wouldn't this be simpler?
```suggestion
    RangeStepStyle(r) isa RangeStepRegular || return _findfirst(p, r)
```

Also maybe use a more explicit name for `_findfirst`, like `_findfirst_general`?

Reply:
Yes this is shorter. I guess I was trying to follow how other traits are treated, but now I have no idea what examples I followed. Maybe there is no need to leave this open to later definition of more methods, `isequal, ::StepRange` is already quite narrow. 

Will see if CI agrees...